powersimdata/input/transform_grid.py

```diff
@@ -356,29 +356,33 @@ def _add_gencost(self):
 
     def _add_storage(self):
         """Adds storage to the grid."""
-        storage_id = self.grid.plant.shape[0]
-        for bus_id, value in self.ct["storage"]["bus_id"].items():
-            storage_id += 1
-            self._add_storage_unit(bus_id, value)
+        first_storage_id = self.grid.plant.index.max() + 1
+        for i, entry in enumerate(self.ct["storage"]):
+            storage_id = first_storage_id + i
+            self._add_storage_unit(entry)
             self._add_storage_gencost()
             self._add_storage_genfuel()
-            self._add_storage_data(storage_id, value)
+            self._add_storage_data(storage_id, entry)
+        # Ensure that int columns remain int after the appends convert them to float
+        storage = self.grid.storage
+        storage["gen"] = storage["gen"].astype({"bus_id": "int", "status": "int"})
```

Comment:
should we move this line into `_add_storage_unit`?

Reply:
Yeah, I think that makes more sense

Reply:
Done.